src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -100,12 +102,14 @@ private static Expression BuildArgumentThatMatchesAnything<TValue>()
 
         private static FakeManager GetFakeManagerCallIsMadeOn(ParsedCallExpression parsedCallExpression)
         {
-            if (parsedCallExpression.CallTarget == null)
-            {
-                throw new ArgumentException("The specified call is not made on a fake object.");
-            }
+            return parsedCallExpression.CallTarget == null
+                ? null

```

Comment:
I'm detecting a code smell here. `CallTo<T>(Expression<Func<T>> callSpecification)` calls this method and then de-reference the return value. Now, that method also calls `AssertThatMemberCanBeIntercepted()` before de-referencing the return value from this method, and I'm guessing that there's no way that method can not throw an exception if the return value from this method is null, hence an NRE is never thrown. However, none of this is obvious to the reader of the code.

Also, it's not obvious that `EnsureCallIsMadeOnAFake` will never be called in a situation where `parsedCallExpression.CallTarget != null`.

I think we should get rid of `GetFakeManagerCallIsMadeOn` and inline the usage of `Fake.GetFakeManager(parsedCallExpression.CallTarget)`.